core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileBatchSource.java

```diff
@@ -134,7 +150,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   public void prepareRun(BatchSourceContext context) throws Exception {
     // Need to create dataset now if macro was provided at configure time
     if (config.timeTable != null && !context.datasetExists(config.timeTable)) {
-      context.createDataset(config.timeTable, KeyValueTable.class.getName(), DatasetProperties.EMPTY);
+      ExploreProperties.Builder builder = ExploreProperties.builder();
+      if (!Strings.isNullOrEmpty(config.timeTableExplorer)) {
+        builder.setExploreTableName(config.timeTableExplorer);
+      } else {
```

Comment:
duplicate code. figure out a way to have this code once and call it multiple times.